quil-py/src/program/mod.rs

```diff
@@ -159,17 +159,22 @@ impl PyProgram {
     }
 
     #[getter]
-    pub fn defined_gates(&self, py: Python<'_>) -> PyResult<Vec<PyGateDefinition>> {
```

Comment:
Since this is already a breaking change, I'm removing this property. It's a vestige from one of the first pushes to implement `quil-py`, and superseded by the new property.